src/components/AppAnalytics/AppAnalytics.tsx

```diff
@@ -6,50 +6,63 @@ import CoolDownButton from "components/Button/CoolDownButton"
 
 const AppAnalytics: React.FC = () => {
   const { getAccessTokenSilently } = useAuth0()
-  const [data, setData] = useState("")
+  const [data, setData] = useState({
+    applicantCount: undefined,
+    firstTime: undefined,
+    ucscApplicants: undefined,
+  })
   const [error, setError] = useState("")
+
   const onClick = async () => {
     const token = await getAccessTokenSilently()
     getAnalytics(token)
       .then(res => {
-        if (res.status !== 201) {
-          setError("Unable to Retrieve")
+        if (res.status === 200 || res.status === 201) {
+          // the server responded successfully, now check our own status code
+          const { message } = res.data
+          if (message === "No Document") {
+            setError("There are no applicants :(")
+          } else {
+            setData(message)
+            setError("")
+          }
         } else {
-          setData("string")
+          setError("Something went wrong!")
         }
       })
       .catch(err => setError(err))
   }
+
   return (
     <div className='app-analytics-component'>
       <div className='app-analytics-component__button'>
         <CoolDownButton
-          className='button__secondary'
+          modifier='secondary'
           duration={30 * 60 * 100}
           label='analytics'
           onClick={onClick}
         >
           Get Analytics
         </CoolDownButton>
-        {error && <div>Unable to Fetch Data</div>}
+        {error && <div className='app-analytics-component__error'>{error}</div>}
       </div>
       <div className='app-analytics-component__results'>
         <div className='app-analytics-component__result'>
           <div className='app-analytics-component__data'>
-            {data !== "" ? data : "X"}{" "}
+            {data.applicantCount ?? "X"}{" "}
```

Comment:
Seems like this works, but I'm curious since I'm not familiar with this notation on how exactly JavaScript is reading it. I'm assuming it's similar to a ternary

Reply:
Yeah, `??` is the [null coalescing](https://indepth.dev/posts/1300/double-question-marks-typescript-3-7-nullish-coalescing) operator, which is pretty cool. It's pretty similar to a ternary, but the difference is below:

```typescript

// displays "X" if count is the empty string
data.applicantCount !== "" ? data.applicantCount : "X"


// displays "X" if count is null or undefined
data.applicantCount ?? "X"

// should be exactly the same as this, since I don't think data.applicantCount would be null
data.applicantCount !== undefined ? data.applicantCount : "X"


// also the || operator
// displays "X" if count is falsy (0, "", null, undefined, etc)
data.applicantCount || "X"
```

In practice, I don't think `data.applicantCount` will ever be `""`. It could be `undefined` (if the key is not found in `data`) or `0` (if there's no applicant for this category). I don't think it'll be `null` either.

`||` and `??` are very similar, except `||` would display `X` if the count is 0 but `??` would just display the number 0. 



I might actually change it to `data.applicantCount !== undefined ? data.applicantCount : "X"` now that I think about it, in case `??` is not supported everywhere.